### DISCUSSION TITLE
How to have multiple group to handle different branches of same repository

### DISCUSSION THREAD

**Discussion by SivaCn** [python-poetry/poetry#8465] [Q&A] [answered]
Hi,

Thanks in advance.

I have multiple private repositories with each having two branches one for dev and another for release. I was trying to configure the below in toml file, and was not successful

Expected
`[tool.poetry.group.dev.dependencies]
abc = {git = "git@abc.git", rev = "develop"}
xyz = {git = "git@xyz.git", rev = "develop"}

[tool.poetry.group.stage.dependencies]
abc = {git = "git@abc.git", rev = "release"}
xyz = {git = "git@xyz.git", rev = "release"}
`

when I ran 
`poetry add git+ssh://abc.git#release --group stage`

had encountered the below error
```
Updating dependencies
Resolving dependencies... (3.0s)

Incompatible constraints in requirements of python-package-manager (0.2.1):
blicore-build-scripts @ git+ssh://abc.git@release (1.1.83)
blicore-build-scripts @ git+ssh://abc.git@develop (1.1.83)
```

any input will be helpful

**Comment by RakeshAMD** [comment]
Need answer for this problem

**Comment by radoering** [comment, marked as answer]
That's not possible. Groups have to be compatible. There is nothing like mutual exclusive groups in Poetry.
